[eos] fix banner when transport is ssh

derekdainys opened the pull request:
When setting transport="ssh" the heredoc commands cause an issue because they are not a string.
This allows for heredoc commands to not be converted when using "ssh" transport

derekdainys committed:
fix banner when transport is ssh

ktbyers commented:
@bewing You know a lot more about this area of the code? Any objections?

bewing commented:
Can we add a SSH test double class to https://github.com/napalm-automation/napalm/blob/develop/test/eos/conftest.py and possibly start testing some of the differences in behavior between the SSH and eAPI transports/drivers/devices?

ktbyers commented:
@bewing Do we want to do this in `conftest.py` or have a separate `test/eos` and `test/eos_ssh` folder.

The latter is probably more consistent with what we did for other drivers (though we also have more code separation at the driver level between `nxos` and `nxos_ssh` (and between `iosxr` and `iosxr_netconf`).

Ultimately, I am fine with either/both (though probably have a preference for separate test folders for the different drivers+transport).

bewing commented:
I also think `eos_ssh` folder makes more sense.  I'll look at cribbing what's necessary from the existing _ssh folders and the standard EOS test double to arrange the boilerplate.  @derekdainys if you can actually collect the data for the unit test it would be great - let me know on slack if you need any guidance

derekdainys committed:
add ssh tests + blank line heredoc test

derekdainys committed:
remove whitespace and comment

derekdainys committed:
black format

derekdainys committed:
remove datetime and re

derekdainys committed:
noqa for linter

derekdainys committed:
reformat heredoc

derekdainys commented:
@bewing  Would like to pick this up again
Noticed that we didnt have a test for heredoc that has blank lines for example in a banner and I added it
This caused me to remove the blank line skip in `_load_config`
Added a `test/eos_ssh` folder which sets a transport=ssh and does some heredoc testing

bewing commented on test/eos_ssh/test_heredoc.py:
```diff
@@ -0,0 +1,190 @@
+import mock
```
should be `from unittest import mock`

https://docs.python.org/3/library/unittest.mock.html

derekdainys replied:
Thanks! Anything else you see?
Some of my builds were passing before but had I had to rebase.

derekdainys committed:
fix mock

derekdainys commented:
bump @bewing @ktbyers

ktbyers committed:
Merge branch 'develop' into eos-ssh-banners

bewing commented:
Apologies for the delays.  Tests look good, much thanks!
